browser/ui/webui/brave_shields/shields_panel_data_handler.cc

```diff
@@ -121,6 +121,9 @@ ShieldsPanelDataHandler::GetActiveShieldsDataController() {
   DCHECK(profile);
 
   Browser* browser = chrome::FindLastActiveWithProfile(profile);
```

Comment:
I'm wondering if it would be better / possible to get the `Browser` that the `webui_controller_` is already associated with via `FindBrowserWithWebContents` but maybe that only works for WebContents on the TabStrip, and not bubbles. Although if we have access to the tab strip, perhaps we could provide it directly to the controller.

Reply:
solved here: https://github.com/brave/brave-core/pull/12984 - we get the right browser in ShieldsPanelUI and cache the TabStripModel